subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/ExposableRepository.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.repositories;
+
+import java.util.Map;
+
+public interface ExposableRepository extends ResolutionAwareRepository {
+
+    Map<RepositoryPropertyType, ?> getProperties();
+
+    enum RepositoryPropertyType {
```

Comment:
This needs a `@UsedByScanPlugin`. While we don't reference the type, we will be expecting stability in these values.

See https://github.com/gradle/gradle/blob/e1bff1e38dc4af9b6037817d722b6ec5b2c70774/subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskOutputCachingDisabledReasonCategory.java#L21 for precedent.

Reply:
done